addon/components/connect.js

```diff
@@ -1,58 +1,57 @@
 import Ember from 'ember';
 
-const { computed, defineProperty, run } = Ember;
+const {
+  computed,
+  defineProperty,
+  inject: { service },
+  isEmpty,
+  run
+} = Ember;
+
+export default (stateToComputed=() => ({}), dispatchToActions=() => ({})) => {
+
+  return Component => {
+
+    return Component.extend({
+
+      redux: service(),
 
-var connect = function(mapStateToComputed, mapDispatchToActions) {
-  var shouldSubscribe = Boolean(mapStateToComputed);
-  var finalMapStateToComputed = mapStateToComputed || function() {return {};};
-  var finalMapDispatchToActions = mapDispatchToActions || function() {return {};};
-  return function wrapWithConnect(WrappedComponent) {
-    var mapState = function(state) {
-      var props = [];
-      Object.keys(finalMapStateToComputed(state)).forEach(function(key) {
-        props.push(key);
-      });
-      return props;
-    };
-    var mapDispatch= function(dispatch) {
-      var actions = [];
-      Object.keys(finalMapDispatchToActions(dispatch)).forEach(function(key) {
-        actions.push(key);
-      });
-      return actions;
-    };
-    return WrappedComponent.extend({
-      redux: Ember.inject.service('redux'),
       init() {
-        var component = this;
-        component['actions'] = Ember.$.extend({}, component['actions']);
-        var redux = this.get('redux');
-        var props = mapState(redux.getState());
-        var dispatch = mapDispatch(redux.dispatch.bind(redux));
-        props.forEach(function(name) {
-          defineProperty(component, name, computed(function() {
-            return finalMapStateToComputed(redux.getState())[name];
-          }).property().readOnly());
-        });
-        dispatch.forEach(function(action) {
-          component['actions'][action] = finalMapDispatchToActions(redux.dispatch.bind(redux))[action];
-        });
-        if (shouldSubscribe && !this.unsubscribe) {
-          this.unsubscribe = redux.subscribe(this.handleChange.bind(this));
+        const redux = this.get('redux');
+
+        let props = stateToComputed(redux.getState());
+
+        for (let name in props) {
```

Comment:
Looking around the web it appears Object.keys is more performant and how this works today. Let's keep these enumerations as Object.keys